packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -81,19 +81,25 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
       client_id: appClientId,
     ));
 
-    // Subscribe to changes in the auth instance returned by init,
-    // and cache the _lastSeenUser as we get notified of new values.
-    final Completer<auth2.GoogleUser> initUserCompleter =
-        Completer<auth2.GoogleUser>();
-
-    auth.currentUser.listen(allowInterop((auth2.GoogleUser nextUser) {
-      if (!initUserCompleter.isCompleted) {
-        initUserCompleter.complete(nextUser);
-      } else {
-        _lastSeenUser = nextUser;
-      }
+    Completer<void> isAuthInitialized = Completer<void>();
+    _isAuthInitialized = isAuthInitialized.future;
+
+    auth.then(allowInterop((auth2.GoogleAuth initializedAuth) {
+      // onSuccess
```

Comment:
Thanks for adding this. I didn't realize this was from generated code so it doesn't have named parameters.

Reply:
Yeah, I think there's a good opportunity for improvement here (auto-wrapping functions in allowInterop, and maybe named parameters).

I'll float the idea around here!

Reply:
I cut a ticket to the tool that we used to generate the code:

* https://github.com/dart-lang/js_facade_gen/issues/78

_(BTW, we can contribute to the tool very easily, WINK, WINK)_